File: Geklaut/05_WTApp_Gutschier/src/main/java/htlvillach/gutschim/wtapp/models/Pupil.java
package htlvillach.gutschim.wtapp.models;

import lombok.Data;

import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.JoinTable;
import javax.persistence.ManyToMany;
import java.util.List;

@Data
@Entity
public class Pupil extends User {


  @ManyToMany
  @JoinTable(
          name = "joins",
          joinColumns = @JoinColumn(name = "pupil_id"),
          inverseJoinColumns = @JoinColumn(name = "timeslot_id"))
  private List<TimeSlot> joins;

}

File: KOTLIN/Wirtschaftstag/WTMobile/app/src/main/java/com/example/wtmobile/data/models/Pupil.kt
package com.example.wtmobile.data.models

import com.example.wtmobile.data.models.TimeSlot

data class Pupil(
    var id: Int,
    var name: String,
    var pwdToken: String,
    var email: String,
    var joins: List<TimeSlot>
)